gnomad/resources/grch38/gnomad.py

```diff
@@ -407,3 +442,122 @@ def release_vcf_path(data_type: str, version: str, contig: str) -> str:
+) -> list:
+    """
+    Perform gnomad GKS annotations on a range of variants at once.
+
+    :param locus_interval: Hail IntervalExpression of locus<reference_genome>.
+        e.g. hl.locus_interval('chr1', 6424776, 6461367, reference_genome="GRCh38")
+    :param version: String of version of gnomAD release to use.
+    :param data_type: String of either "exomes" or "genomes" for the type of reads that are desired.
+    :param by_ancestry_group: Boolean to pass to obtain frequency information for each cohort.
+    :param by_sex: Boolean to pass to return frequency information for each cohort split by chromosomal sex.
+    :param vrs_only: Boolean to pass for only VRS info to be returned
+        (will not include allele frequency information).
+    :param custom_ht: Table to use instead of what public_release() method would return for the version.
+    :param skip_coverage: Bool to pass to skip adding coverage statistics.
+    :param custom_coverage_ht: Custom table to use for coverage statistics instead of the release coverage table.
+    :return: List of dictionaries containing VRS information
+        (and freq info split by ancestry groups and sex if desired) for specified variant.
+    """
+    # Read public_release table if no custom table provided
+    if custom_ht:
+        ht = custom_ht
+    else:
+        ht = hl.read_table(public_release(data_type).versions[version].path)
+
+    high_level_version = f"v{version.split('.')[0]}"
+
+    # Read coverage statistics if requested
+    if high_level_version == "v3":
+        coverage_version = "3.0.1"
+    else:
+        raise NotImplementedError(
+            "gnomad_gks() is currently only implemented for gnomAD v3."
+        )
+
+    coverage_ht = None
+
+    if not skip_coverage:
+        if custom_coverage_ht:
+            coverage_ht = custom_coverage_ht
+        else:
+            coverage_ht = hl.read_table(
+                coverage("genomes").versions[coverage_version].path
+            )
+
+    # Retrieve ancestry groups from the imported POPS dictionary.
+    pops_list = list(POPS[high_level_version]) if by_ancestry_group else None
+
+    # Throw warnings if contradictory arguments are passed.
+    if by_ancestry_group and vrs_only:
+        logger.warning(
+            "Both 'vrs_only' and 'by_ancestry_groups' have been specified. Ignoring"
+            " 'by_ancestry_groups' list and returning only VRS information."
+        )
+    elif by_sex and not by_ancestry_group:
+        logger.warning(
+            "Splitting whole database by sex is not yet supported. If using 'by_sex',"
+            " please also specify 'by_ancestry_group' to stratify by."
+        )
+
+    # Call and return add_gks_vrs and add_gks_va for chosen arguments.
+
+    # Select relevant fields, checkpoint, and filter to interval before adding
+    # annotations
+    ht = ht.select(ht.freq, ht.info.vrs, ht.popmax)
+    ht = ht.checkpoint(hl.utils.new_temp_file("vrs_checkpoint", extension="ht"))
```

Comment:
If I can make one last suggestion, could there be an optional argument like `disable_checkpoint` to `gnomad_gks` that turns off this checkpoint, and a doc string that says it is recommended to leave on for larger datasets?

I'm just observing that for small tables that are more like tens of thousands of variants and <1GB in size and can comfortably fit in memory, the checkpoint adds a lot of time overhead.

Reply:
Thats definitely possible. We are also about to make one more change for Popmax (where AF =/= FAF) so yours isn't the last suggestion:)

Reply:
Awesome, thanks @matren395

Reply:
@matren395 I can push a branch with that and create another multi-tiered PR